### DISCUSSION TITLE
How can I customize the display items and SQL in AttachAction?

### DISCUSSION THREAD

**Discussion by bzy107** [filamentphp/filament#9181] [Help]
I've read the following document. 
However, can customization of the select box and SQL during attachment be done for relationships other than belongsTo?

https://filamentphp.com/docs/3.x/panels/resources/relation-managers#attaching-and-detaching-records

**Comment by sinnbeck** [comment]
Yes it works for `BelongsToMany` and `MorphToMany`

**Reply by bzy107** [reply]
Is it possible to customize the search using something other than `whereBelongsTo`, as shown here:
[https://filamentphp.com/docs/3.x/panels/resources/relation-managers#scoping-the-options-to-attach](https://filamentphp.com/docs/3.x/panels/resources/relation-managers#scoping-the-options-to-attach)
```
AttachAction::make()
    ->recordSelectOptionsQuery(fn (Builder $query) => $query->whereBelongsTo(auth()->user())
```
